templates/dronerunner/index.ts

```diff
@@ -0,0 +1,106 @@
+import { Services, createTemplate } from "~templates-utils";
+
+export default createTemplate({
+  name: "Drone Runner",
+  meta: {
+    description: "Runner for Drone.io",
+    changeLog: [{ date: "2022-08-03", description: "first release" }],
+    links: [
+      { label: "Website", url: "https://drone.io/" },
+      {
+        label: "Documentation",
+        url: "https://github.com/harness/drone#setup-documentation",
+      },
+      { label: "Github", url: "https://github.com/harness/drone" },
+    ],
+    contributors: [
+      { name: "Ivan Ryan", url: "https://github.com/ivanonpc-22" },
+    ],
+  },
+  schema: {
+    type: "object",
+    required: ["projectName", "domain", "serviceName", "rpcHost", "rpcSecret", "runnerCapacity"],
+    properties: {
+      projectName: {
+        type: "string",
+        title: "Project Name",
+      },
+      domain: {
+        type: "string",
+        title: "Domain",
+      },
+      serviceName: {
+        type: "string",
+        title: "App Service Name",
+        default: "drone-runner",
+      },
+      rpcHost: {
+        type: "string",
+        title: "Drone Server Hostname",
+        default: "drone.company.com",
+      },
+      rpcSecret: {
+        type: "string",
+        title: "RPC Secret",
+        default: "Secret",
+      },
+      RPCprotocol: {
+        type: "string",
+        title: "RPC Protocol",
+        default: "https",
+        oneOf: [
+          { enum: ["https"], title: "https" },
+          { enum: ["http"], title: "http" },
+        ],
+      },
+      runnerCapacity: {
+        type: "string",
+        title: "Runner Capacity",
+        default: "2",
+      },
+    },
+  } as const,
+    generate({
+      projectName,
+      domain,
+      serviceName,
+      rpcHost,
+      rpcSecret,
+      RPCprotocol,
+      runnerCapacity,
+    }) {
+      const services: Services = [];
+
+      services.push({
+        type: "app",
+        data: {
+          projectName,
+          serviceName: serviceName,
+          env: [
+            `DRONE_RPC_HOST=${rpcHost}`,
+            `DRONE_RPC_PROTO=${RPCprotocol}`,
+            `DRONE_RUNNER_CAPACITY=${runnerCapacity}`,
+            `DRONE_RPC_SECRET=${rpcSecret}`,
+          ].join("\n"),
+          source: {
+            type: "image",
+            image: "drone/drone-runner-docker:1",
+          },
+          proxy: {
+            port: 3000,
+            secure: true,
+          },
+          domains: [{ name: domain }],
+          mounts: [
+            {
+            type: "bind",
+            hostPath: "/var/run/docker.sock",
+            mountPath: "/var/run/docker.sock",
```

Comment:
Formating is off. Please install prettier.

Reply:
Lol that is my bad, I switched linux operating systems very recently and didn't get my IDE and programs setup all the way.

Will fix that right now.